src/lib/converter/plugins/GitHubPlugin.ts

```diff
@@ -82,9 +90,13 @@ class Repository {
             });
         }
 
-        out = <ShellJS.ExecOutputReturnValue> ShellJS.exec('git rev-parse --short HEAD', {silent: true});
-        if (out.code === 0) {
-            this.branch = out.stdout.replace('\n', '');
+        if (!this.revision) {
+            out = <ShellJS.ExecOutputReturnValue> ShellJS.exec('git rev-parse --short HEAD', {silent: true});
+            if (out.code === 0) {
+                this.branch = out.stdout.replace('\n', '');
+            }
+        } else if (this.revision) {
+            this.branch = this.revision;
```

Comment:
If you feel `revision` more accurately reflects what it is, let's use that. I agree that branch is the least useful. As for the CLI flag, can we maybe do `--git-revision` to make it less ambiguous? I feel like any of `revision`, `reference` or `branch` without context won't make a ton of sense.